lib/acpi/acpi.c

```diff
@@ -455,6 +456,107 @@ ACPI_RESOURCE *acpi_resource_parse(ACPI_RESOURCE *res, int res_type)
 	return res;
 }
 
+int acpi_device_irq_get(struct acpi_dev *child_dev, struct acpi_irq_resource *irq_res)
+{
+	ACPI_RESOURCE *res = acpi_resource_parse(child_dev->res_lst, ACPI_RESOURCE_TYPE_IRQ);
+
+	if (!res) {
+		res = acpi_resource_parse(child_dev->res_lst, ACPI_RESOURCE_TYPE_EXTENDED_IRQ);
+		if (!res) {
+			return -ENODEV;
+		}
+
+		if (res->Data.ExtendedIrq.InterruptCount > CONFIG_ACPI_IRQ_VECTOR_MAX) {
+			return -ENOMEM;
+		}
+
+		memset(irq_res, 0, sizeof(struct acpi_irq_resource));
+		irq_res->irq_vector_max = res->Data.ExtendedIrq.InterruptCount;
+		for (int i = 0; i < irq_res->irq_vector_max; i++) {
+			irq_res->irqs[i] = (uint16_t)res->Data.ExtendedIrq.Interrupts[i];
+		}
+
+		irq_res->flags = arch_acpi_encode_irq_flags(res->Data.ExtendedIrq.Polarity,
+							    res->Data.ExtendedIrq.Triggering);
+	} else {
+		if (res->Data.Irq.InterruptCount > CONFIG_ACPI_IRQ_VECTOR_MAX) {
+			return -ENOMEM;
+		}
+
+		irq_res->irq_vector_max = res->Data.Irq.InterruptCount;
+		for (int i = 0; i < irq_res->irq_vector_max; i++) {
+			irq_res->irqs[i] = (uint16_t)res->Data.Irq.Interrupts[i];
+		}
+
+		irq_res->flags = arch_acpi_encode_irq_flags(res->Data.ExtendedIrq.Polarity,
+							    res->Data.ExtendedIrq.Triggering);
+	}
+
+	return 0;
+}
+
+int acpi_device_mmio_get(struct acpi_dev *child_dev, struct acpi_mmio_resource *mmio_res)
+{
+	ACPI_RESOURCE *res = child_dev->res_lst;
+	struct acpi_reg_base *reg_base = mmio_res->reg_base;
+	int mmio_cnt = 0;
+
+	memset(mmio_res, 0, sizeof(*mmio_res));
+
+	do {
+		if (!res->Length) {
+			LOG_DBG("Error: zero length found!");
```

Comment:
If this an error (like the log message suggests) why are you using `LOG_DBG()` instead of `LOG_ERR()`?

Reply:
> There might be device/node with no resource and hence given as only debug info.

Then I'd suggest to remove "Error:" from the message, since that makes it look like an actual error rather than debug info.